### PR TITLE
[4.0] Add upload file name extension check to com_installer's Upload & Install function (1st attempt)

### PR DESCRIPTION
**Pull request by richard67**
Pull Request for Issue [https://github.com/joomla/joomla-cms/issues/29763#issuecomment-647978807](https://github.com/joomla/joomla-cms/issues/29763#issuecomment-647978807) (at least partly).

### Summary of Changes

This Pull Request (PR) adds a check of the file name extension (or suffix) to the Upload & Install function of the Extensions Installer.

Currently only zip files work with Upload & Install, and so this PR will only allow files with extension (or suffix) ".zip" (case insensitive). If later more formats (".tar.gz", ".tar.bz2", ...) will be supported, the regular expression used for this new check can be easily extended by the new suffixes (has to be done at 2 places), or a server side check could check for the availability of the particular PHP modules and pass the regular expression to the js e.g. with a `data-` attribute or a hidden input field.

**Important:** This is **NOT** a security fix, it only shall prevent from accidently selecting the wrong file for upload and then having to wait until the upload completes just for getting an error message which is not really user friendly when unpacking the file fails. Further server-side checks, e.g. by mime type or checks of the content) have to be implemented separate from this PR with some future PR(s).

### Testing Instructions

Will be added soon. Until this has been done I will leave this PR in draft status. As soon as draft status will be removed, the PR can be tested.

### Actual result BEFORE applying this Pull Request

Will be added soon.

### Expected result AFTER applying this Pull Request

Will be added soon.

### Documentation Changes Required

None.

### CODE REVIEW

File: administrator/language/en-GB/com_installer.ini [hunk 1]
```diff
@@ -193,6 +193,8 @@ COM_INSTALLER_MSG_WARNINGS_SMALLUPLOADSIZEDESC="The maximum file size for upload
 COM_INSTALLER_MSG_WARNINGS_UPDATE_NOTICE="Before updating ensure that the update is compatible with your Joomla! installation. <br>You are strongly advised to make a <strong>backup</strong> of your installation before you start updating."
 COM_INSTALLER_MSG_WARNINGS_UPLOADBIGGERTHANPOST="PHP Upload Size bigger than POST size."
 COM_INSTALLER_MSG_WARNINGS_UPLOADBIGGERTHANPOSTDESC="The value of the upload_max_filesize in the php.ini file is greater than the post_max_size variable. The post_max_size variable will take precedence and block requests larger than it. This is generally a server misconfiguration when trying to increase upload sizes. Please increase the upload_max_filesize to at least match the post_max_size variable or vice versa."
+; Please do not translate '.zip' in the following language string
+COM_INSTALLER_MSG_WARNINGS_UPLOADFILESUFFIX="The selected file cannot be uploaded as it has a suffix different to \".zip\"."
```
Comment (brianteeman) [inline administrator/language/en-GB/com_installer.ini:197]: You do realise that we offer updates in more formats than zip?

Comment (brianteeman) [reply administrator/language/en-GB/com_installer.ini:197]: sorry just read your comment on this.

Comment (richard67) [reply administrator/language/en-GB/com_installer.ini:197]: @brianteeman Yes, but they can be only used for the "Install from folder" method. The Upload method only works with zip.

**Comment by richard67** [comment]
It seems the issue tracker doesn't recognize this PR as a PR but thinks it is an issue. Test buttons are missing. I close it and make it again with a new PR.

**Pull request closed without merging**